auth/certificate_authenticator.hh

```diff
@@ -10,7 +10,7 @@
 #pragma once
 
 #include "auth/authenticator.hh"
-#include <boost/regex_fwd.hpp>
+#include <boost/regex_fwd.hpp>  // IWYU pragma: keep
```

Comment:
This is sad, reminds me of the ugly "noqa" lines in dtest (578 such lines in dtest, on my last count) 
Isn't this a sign we should stop using the buggy "include-cleaner"? Does it bring us any benefit?
Even the comment's name, "IWYU" ("include what you use") is a wrong - since the problem being detected here wasn't that we weren't including what we were using, but the opposite problem - that (supposedly) we included something we didn't need.

@avikivity what is your opinion on this include-cleaner thing? Is it something you wanted or like?

Reply:
I'm not a fan of it. But since most of the contributors aren't concerned with dependency reduction, maybe it helps a little.